### DISCUSSION TITLE
Ability to choose a table prefix to allow for multi project schema

### DISCUSSION THREAD

**Discussion by MendyLanda** [payloadcms/payload#5590] [Feature Requests & Ideas]
Normally if drizzle is fully available, something like this will be possible:
```ts
export const createTable = pgTableCreator((name) => `projectX_${name}`);
```
This in conjunction with the filter in the drizzle config, will allow developers to have multiple projects in the same db schema but keep them separated to each project.

From looking at payload's source code, this will be a relatively simple implementation and I will be happy to make a PR if you want to have this feature. 

For more details:
https://orm.drizzle.team/docs/goodies#multi-project-schema

**Comment by DanRibbens** [comment]
This is a great suggestion! I think this will be very easy to add with our change to custom table names https://github.com/payloadcms/payload/pull/5045. In this branch, all the table names are assigned from a new function so this change is only going to be a few lines of code. 

I can get this added after that is merged. It had a few errors in testing that just need a little tweaking before merging and releasing.

Thanks for the suggestion!

**Reply by MendyLanda** [reply]
Awesome 👏 thanks 🙏 

p.s. I later realized some work was being done in this area in pr #4525 so I left a [comment](https://github.com/payloadcms/payload/pull/4525#issuecomment-2031923450) there also.

**Comment by 0xRaini** [comment]
This memory system approach is really interesting! 🧠

I just published a skill called **Soulmate** that takes a similar concept but applies it specifically to relationship/companion scenarios. It tracks:

- **Intimacy levels** that persist across sessions
- **Streak tracking** (consecutive days of interaction)
- **Unlocked personas** based on interaction patterns
- **Achievements** triggered by specific behaviors

The state is stored in a simple JSON file in the workspace, but the weighting approach you describe could be really powerful for making the relationship feel more "real" - remembering important moments vs casual chat.

Would love to see how your memory entities could integrate with emotional context. For example: remembering that the user was sad last Tuesday, and following up on it.

If anyone wants to try it: `clawhub install soulmate`
